dotty-tensorflow/src/main/scala/me/shadaj/scalapy/tensorflow/api/keras/models/Sequential.scala

```diff
@@ -23,7 +25,7 @@ class Sequential private[api] (val underlying: PySequential) extends PythonType[
       targetTensors: Option[String] = None
   ) =
     underlying.compile(
-      fromPyhtonTypeAndScalaTypeUnion(optimizer),
+      fromEnumPythonTypeUnion(optimizer),
```

Comment:
Hmm in general, which of our `from...` methods are ever used as implicit conversions? Maybe replace them with extension methods/implicit classes instead: `optimizer.toScalaTypeUnion` (or sth like that)?

Reply:
Ok small analysis:
| name                      | explicit | implicit |
|---------------------------|----------|----------|
| `fromEnumPythonTypeUnion` | 0        | 1        |
| `TensorToPyTensor`        | 0        | 19       |
| `PyTensorToTensor`        | 0        | 15       |
| `fromPythonUnion`         | 1        | 3        |
| `fromScalaTypesUnion`     | 1        | 4        |
| `fromPythonTypeSeqsUnion` | 0        | 2        |
| `option2PyOption`         | 3        | 39       |
| `toScalaOption`           | 5        | 14       |
| `enum2String`             | 2        | 12       |

Those that had 0 usages were removed.
1. There seem to be a problem when creating implicit vals so I turned all of them into defs
2. Extension methods are hard :/
3. As you can see we need them to remain implicit mostly :D

Reply:
Wow how you did you count the above? I mean, explicit can be Ctrl+F-ed... but can VSCode reliably count the implicit uses?

Reply:
you can just comment out and compile. 
that way you get all errors of wrong type (implicit) and unrecognised function (explicit). 
It's more or less accurate